src/main/java/org/folio/inventory/dataimport/consumers/MarcBibUpdateKafkaHandler.java

```diff
@@ -0,0 +1,90 @@
+package org.folio.inventory.dataimport.consumers;
+
+import static java.lang.String.format;
+import static org.folio.rest.util.OkapiConnectionParams.OKAPI_TENANT_HEADER;
+import static org.folio.rest.util.OkapiConnectionParams.OKAPI_TOKEN_HEADER;
+import static org.folio.rest.util.OkapiConnectionParams.OKAPI_URL_HEADER;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.MappingMetadataDto;
+import org.folio.dbschema.ObjectMapperTool;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.dataimport.cache.MappingMetadataCache;
+import org.folio.inventory.dataimport.handlers.actions.InstanceUpdateDelegate;
+import org.folio.inventory.dataimport.handlers.matching.util.EventHandlingUtil;
+import org.folio.kafka.AsyncRecordHandler;
+import org.folio.kafka.KafkaHeaderUtils;
+import org.folio.processing.exceptions.EventProcessingException;
+import org.folio.rest.jaxrs.model.Event;
+import org.folio.rest.jaxrs.model.Record;
+
+public class MarcBibUpdateKafkaHandler implements AsyncRecordHandler<String, String> {
+
+  private static final Logger LOGGER = LogManager.getLogger(MarcBibUpdateKafkaHandler.class);
+  private static final String MAPPING_METADATA_NOT_FOUND_MSG = "MappingParameters and mapping rules snapshots were not found by jobExecutionId '%s'";
+  private static final String RECORD_KEY = "record";
+  private static final String MAPPING_RULES_KEY = "MAPPING_RULES";
+  private static final String MAPPING_PARAMS_KEY = "MAPPING_PARAMS";
+  private static final ObjectMapper OBJECT_MAPPER = ObjectMapperTool.getMapper();
+  private static final String RECORD_ID_HEADER = "recordId";
+  private static final String CHUNK_ID_HEADER = "chunkId";
+  private static final String JOB_ID = "jobId";
+  private static final String MARC_BIB_RECORD_TYPE = "marc-bib";
+
+  private final InstanceUpdateDelegate instanceUpdateDelegate;
+  private final MappingMetadataCache mappingMetadataCache;
+
+  public MarcBibUpdateKafkaHandler(InstanceUpdateDelegate instanceUpdateDelegate, MappingMetadataCache mappingMetadataCache) {
+    this.instanceUpdateDelegate = instanceUpdateDelegate;
+    this.mappingMetadataCache = mappingMetadataCache;
+  }
+
+  @Override
+  public Future<String> handle(KafkaConsumerRecord<String, String> record) {
+    try {
+      Promise<String> promise = Promise.promise();
+      Event event = OBJECT_MAPPER.readValue(record.value(), Event.class);
+      @SuppressWarnings("unchecked")
+      HashMap<String, String> eventPayload = OBJECT_MAPPER.readValue(event.getEventPayload(), HashMap.class);
```

Comment:
You should create a class that will describe the format of our event and then deserialize it there.